### PR TITLE
(Fixes #467) Handle InvalidRequestException properly

### LINKED ISSUE #467: InvalidRequestException is not handled in the create and upload pipeline transaction
The following exception is thrown but the pipeline is not cleaned up

```
Exception in thread "main" com.amazonaws.services.datapipeline.model.InvalidRequestException: Web service limit exceeded: Exceeded maximum number of objects allowed per pipeline (Service: DataPipeline; Status Code: 400; Error Code: InvalidRequestException; Request ID: 96f55510-933c-11e6-ab00-9b48e5f8ec63)
	at com.amazonaws.http.AmazonHttpClient.handleErrorResponse(AmazonHttpClient.java:1275)
	at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:873)
	at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:576)
	at com.amazonaws.http.AmazonHttpClient.doExecute(AmazonHttpClient.java:362)
	at com.amazonaws.http.AmazonHttpClient.executeWithTimer(AmazonHttpClient.java:328)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:307)
	at com.amazonaws.services.datapipeline.DataPipelineClient.invoke(DataPipelineClient.java:1479)
	at com.amazonaws.services.datapipeline.DataPipelineClient.putPipelineDefinition(DataPipelineClient.java:1004)
	at com.krux.hyperion.client.UploadPipelineObjectsTrans$$anonfun$2.apply(UploadPipelineObjectsTrans.scala:45)
	at com.krux.hyperion.client.UploadPipelineObjectsTrans$$anonfun$2.apply(UploadPipelineObjectsTrans.scala:45)
	at com.krux.hyperion.client.Retry$Retryable.retry(Retry.scala:23)
	at com.krux.hyperion.client.UploadPipelineObjectsTrans.com$krux$hyperion$client$UploadPipelineObjectsTrans$$createAndUploadObjects(UploadPipelineObjectsTrans.scala:51)
	at com.krux.hyperion.client.UploadPipelineObjectsTrans$$anonfun$action$1.apply(UploadPipelineObjectsTrans.scala:79)
	at com.krux.hyperion.client.UploadPipelineObjectsTrans$$anonfun$action$1.apply(UploadPipelineObjectsTrans.scala:78)
```

### PR DESCRIPTION
**Pull request by realstraw**

### CODE REVIEW

File: core/src/main/scala/com/krux/hyperion/client/UploadPipelineObjectsTrans.scala [hunk 1]
```diff
@@ -43,39 +43,46 @@ case class UploadPipelineObjectsTrans(
...
+
+      putDefinitionResult.getValidationErrors.asScala
+        .flatMap(err => err.getErrors.asScala.map(detail => s"${err.getId}: $detail"))
+        .foreach(log.error)
+      putDefinitionResult.getValidationWarnings.asScala
+        .flatMap(err => err.getWarnings.asScala.map(detail => s"${err.getId}: $detail"))
+        .foreach(log.warn)
+
+      if (putDefinitionResult.getErrored) {
+        log.error("Failed to create pipeline")
```
Comment (prashuj) [inline core/src/main/scala/com/krux/hyperion/client/UploadPipelineObjectsTrans.scala]: isn't the below 2 error messages confusing?

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/client/UploadPipelineObjectsTrans.scala]: It is confusing, let me update

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/client/UploadPipelineObjectsTrans.scala]: fixed

**Comment by prashuj** [comment]
👍

**Pull request merged**